lib/beacon/content.ex

```diff
@@ -20,6 +20,31 @@ defmodule Beacon.Content do
   alias Beacon.Types.Site
   alias Ecto.Changeset
 
+  defp validate_page_template(changeset) do
+    format = Changeset.get_field(changeset, :format)
+    template = Changeset.get_field(changeset, :template)
+    do_validate_page_template(changeset, format, template)
+  end
+
+  defp do_validate_page_template(changeset, :heex = _format, template) when is_binary(template) do
+    site = Changeset.get_field(changeset, :site)
+    path = Changeset.get_field(changeset, :path)
+    metadata = %Beacon.Template.LoadMetadata{site: site, path: path}
+
+    case Beacon.Template.HEEx.compile(template, metadata) do
+      {:cont, _ast} ->
+        changeset
+
+      {:halt, %{description: description}} ->
+        Changeset.add_error(changeset, :template, description)
+
+      {:halt, _} ->
+        Changeset.add_error(changeset, :template, "invalid template")
+    end
```

Comment:
Perhaps this function should return a tagged tuple?  i.e. for error cases `{:error, changeset}` otherwise `{:ok, changeset}`

That way our later `with` blocks would not require `else` clause